cool.py

```diff
@@ -0,0 +1,294 @@
+    row_num = int(row)
+
+    #: download primary tile (top-left)
+    img_1 = requests.get(f"{BASE_URL}/{col_num}/{row_num}").content
+    #: dowload top-right tile
+    img_2 = requests.get(f"{BASE_URL}/{col_num + 1}/{row_num}").content
+    #: dowload bottom-left tile
+    img_3 = requests.get(f"{BASE_URL}/{col_num}/{row_num + 1}").content
+    #: dowload bottom-right tile
+    img_4 = requests.get(f"{BASE_URL}/{col_num + 1}/{row_num + 1}").content
+
+    tile_list = [img_1, img_2, img_3, img_4]
+
+    return tile_list
+
+
+def build_mosaic_image(tiles, col, row, out_dir):
+    """build a mosaic image from a list of cv2 images
+
+    Args:
+        images (list): list of cv2 images to mosaic together
+        col (str): the column of the WMTS index for the tile of interest (top-left tile)
+        row (str): the row of the WMTS index for the tile of interest (top-left tile)
+        out_dir (Path): location to save the result
+
+    Returns:
+        mosaic_image (np.ndarray): composite mosaic of smaller images
+    """
+    
+    tile_name = f'{col}_{row}'
+
+    if tiles is None or len(tiles) == 0:
+        logging.info("no images to mosaic for %s", tile_name)
+
+        return np.array(None)
+
+    #: Set up parameters for images, mosaic, number of cols/rows (every image will be 256x256)
+    tile_width = 256
+    number_columns = 2
+    number_rows = 2
+
+    #: Build mosaic image with white background
+    total_height = tile_width * number_rows
+    total_width = tile_width * number_columns
+
+    logging.info("mosaicking images for %s", tile_name)
+
+    mosaic_image = np.zeros((total_height, total_width, 3), dtype=np.uint8)
+    mosaic_image[:, :] = (255, 255, 255)
+
+    i = 0
+    for tile in tiles:
+        #: convert from bytes to cv2
+        img = convert_to_cv2_image(tile)
+        #: Add image into the mosaic
+        row_start = (math.floor(i / number_columns)) * tile_width
+        col_start = (i % number_columns) * tile_width
+        mosaic_image[row_start : row_start + tile_width, col_start : col_start + tile_width] = img
+
+        i += 1
+
+    if out_dir:
+        if not out_dir.exists():
+            out_dir.mkdir(parents=True)
+
+        mosaic_outfile = out_dir / f"{tile_name}.jpg"
+        logging.debug("saving to %s", mosaic_outfile)
+        cv2.imwrite(str(mosaic_outfile), mosaic_image)
+
+        return mosaic_image
+
+    else:
+        logging.debug("no output directory provided")
+
+        return mosaic_image
+
+
+def load_pytorch_model():
+    """load pytorch model with tower scout weights
+
+    Args:
+        None
+
+    Returns:
+        pytorch model: model ready for scanning
+    """
+    model_weight_path = Path(__file__).parent / 'tower_scout' / 'xl_250_best.pt'
+    yolov5_path = Path(__file__).parent / 'yolov5'
+    model = torch.hub.load(str(yolov5_path), 'custom', path=str(model_weight_path), source='local')
+    #: if statements (check that .pt file exists, folders, etc.)
```

Comment:
**suggestion**: yup. get defensive here in case people didn't run the prerequisites.